pkg/kubelet/cm/dra/types.go

```diff
@@ -17,35 +17,22 @@ package dra
 
 import (
 	v1 "k8s.io/api/core/v1"
-	"k8s.io/kubernetes/pkg/kubelet/cm/topologymanager"
 	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
 )
 
-// ActivePodsFunc is a function that returns a list of pods to reconcile.
-type ActivePodsFunc func() []*v1.Pod
-
 // Manager manages all the DRA resource plugins running on a node.
 type Manager interface {
-	// Allocate prepares and assigns resources to a container in a pod. From
-	// the requested resources, Allocate will communicate with the
-	// DRA resource plugin to prepare resources.
-	Allocate(pod *v1.Pod, container *v1.Container) error
-
-	// TopologyManager HintProvider provider indicates the DRA Manager implements the Topology Manager Interface
-	// and is consulted to make Topology aware resource alignments
-	GetTopologyHints(pod *v1.Pod, container *v1.Container) map[string][]topologymanager.TopologyHint
-
-	// TopologyManager HintProvider provider indicates the DRA Manager implements the Topology Manager Interface
-	// and is consulted to make Topology aware resource alignments per Pod
-	GetPodTopologyHints(pod *v1.Pod) map[string][]topologymanager.TopologyHint
-
-	// GetCDIAnnotations checks whether we have cached resource
-	// for the passed-in <pod, container> and returns its container annotations or
-	// empty map if resource is not cached
-	GetCDIAnnotations(pod *v1.Pod, container *v1.Container) ([]kubecontainer.Annotation, error)
+	// PrepareResources prepares resources for a container in a pod.
+	// It communicates with the DRA resource plugin to prepare resources and
+	// returns container options to trigger CDI injection on the runtime side.
+	PrepareResources(pod *v1.Pod, container *v1.Container) (*DynamicResourceContainerOptions, error)
 
 	// UnprepareResources calls NodeUnprepareResource GRPC from DRA plugin to unprepare pod resources
 	UnprepareResources(pod *v1.Pod) error
 }
 
-const DRACheckpointDir = "/var/lib/kubelet/dra-plugins"
+// DynamicResourceContainerOptions contains runtime settings to consume prepared resources.
+type DynamicResourceContainerOptions struct {
+	// The Annotations for the container
+	Annotations []kubecontainer.Annotation
+}
```

Comment:
The name for this struct seems off somehow. The use of `Options` seems wrong. Need to think what a more appropriate name would be though.

Reply:
Renamed to DRAContainerInfo. Please, propose better name if you don't like this one.